### ISSUE TITLE
Broken install if django is not installed (0.1.3)

### ISSUE THREAD

**Issue by golgor** [a1tus/django-admin-cursor-paginator#6]
Thanks for great work with the django-admin-cursor-paginator!

I have a project where we use it and I ran into an issue with the latest version. We are using it to deploy a django server in a Docker-container and using `pip install -r requirements.txt` to install all our dependencies.

When I tried to upgrade some packages and build our Docker image, the build failed with the following:
```bash
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [55 lines of output]
      Traceback (most recent call last):
        File "/home/golgor/.pyenv/versions/3.8.16/envs/backend_test/lib/python3.8/site-packages/setuptools/config.py", line 387, in _parse_attr
          return getattr(StaticModule(module_name), attr_name)
        File "/home/golgor/.pyenv/versions/3.8.16/envs/backend_test/lib/python3.8/site-packages/setuptools/config.py", line 24, in __init__
          spec = importlib.util.find_spec(name)
        File "/home/golgor/.pyenv/versions/3.8.16/lib/python3.8/importlib/util.py", line 94, in find_spec
          parent = __import__(parent_name, fromlist=['__path__'])
        File "/tmp/pip-install-tygxg39s/django-admin-cursor-paginator_3f4247dc51c54a0fb8b77c6734011856/admin_cursor_paginator/__init__.py", line 1, in <module>
          from .admin import CursorPaginatorAdmin  # noqa: F403
        File "/tmp/pip-install-tygxg39s/django-admin-cursor-paginator_3f4247dc51c54a0fb8b77c6734011856/admin_cursor_paginator/admin.py", line 1, in <module>
          import django
      ModuleNotFoundError: No module named 'django'
```
We have previously used version 0.1.1 without any issues and I ran tests with 0.1.2 as well and it worked fine. 0.1.3 fails. I looked into it and I noticed the following in your setup.cfg:

```
version = attr: admin_cursor_paginator.version.__version__
```

I assume, trying to read the version from admin_cursor_paginator.version means it will import the `admin_cursor_paginator`-package, and in the `__init__.py` you import CursorPaginatorAdmin from the admin.py, which means it imports that module. I noticed the added `import django `in this file which then will try to import the django-package during install. My suspicion is that this is the issue.

I'm not entirely sure why you import the CursorPaginatorAdmin like that in the __init__.py.  Is this a requirement from being a django-app?

The obvious solution would be just add the version in the setup.cfg instead of in the version-module, but I guess there are other work-arounds to fix this.

Edit:
It seems this does not always occur. So far I managed to reproduce the error if I create a new virtual environment (pyenv in my case) or I use `pip cache purge` to clear out the cache. If django have been installed before (and is stored in some cache) it seems to able to solve the dependency, but in case of a completely new environment the install fails.

Edit2:
I tried to remove the `admin_cursor_paginator.version` and define the version 0.1.3 in the setup.cfg directlly and it solved the issue.

**Comment by a1tus** [comment]
Hi, thank you for the investigation. I've planned to add django 4.2 support, so I think I will combine it with this fix (need some time to verify) and release 0.1.4 then.

**Comment by golgor** [comment]
> Hi, thank you for the investigation. I've planned to add django 4.2 support, so I think I will combine it with this fix (need some time to verify) and release 0.1.4 then.

Sounds great! :)

**Comment by a1tus** [comment]
@golgor, hi again! The only difference that I've found between 0.1.1, 0.1.2 (that worked as you say) and 0.1.3 that could somehow cause this effect if `import django` statement.

But it's strange, because `from django.contrib import admin` is there from the first version.

I can't reproduce this, so can you please check locally if moving `import django` inside `get_paginator` will fix it?

**Comment by golgor** [comment]
> @golgor, hi again! The only difference that I've found between 0.1.1, 0.1.2 (that worked as you say) and 0.1.3 that could somehow cause this effect is `import django` statement.
> 
> But it's strange, because `from django.contrib import admin` is there from the first version.
> 
> I can't reproduce this, so can you please check locally if moving `import django` inside `get_paginator` will fix it?

Sure thing, I will take a look again. I was only able to reproduce this when I tried to install this package in a completely empty Virtual Environment or during a Docker build using the Python 3.8-slim.

**Comment by a1tus** [comment]
@golgor, after some further investigation, I've ended up with some simple solution and released 0.1.4, please let me know if everything is fine now (or reopen).

**Comment by golgor** [comment]
@a1tus 
I tested to updated form 0.1.2 -> 0.1.4 and my docker build succeeded, which it didn't with 0.1.3. Thanks for the fix!

Also upgrading to Django 4.2 soon, so your fix/update came perfectly timed. :)
